doc/build-netbsd.md

```diff
@@ -1,81 +1,161 @@
+
+Install the required dependencies the usual way you [install software on NetBSD](https://www.netbsd.org/docs/guide/en/chap-boot.html#chap-boot-pkgsrc) - either with `pkg_add` or `pkgin`. The example commands below use `pkgin` which is the [recommended](https://www.netbsd.org/docs/guide/en/chap-boot.html#chap-boot-pkgsrc) way to install binary packages.
+
+Note: `pkgin` is usually run as `root` or via `sudo`.
+
+```bash
+pkgin install autoconf automake libtool pkg-config git gmake boost libevent
 
 ```
-autoconf
-automake
-boost
-git
-gmake
-libevent
-libtool
-pkg-config
-python37
 
+### 2. Clone Bitcoin Repo
+
+Now that `git` and all the required dependencies are installed, let's clone the Bitcoin Core repository to a directory. All build scripts and commands will run from this directory.
+
+```bash
 git clone https://github.com/bitcoin/bitcoin.git
 ```
 
-See [dependencies.md](dependencies.md) for a complete overview.
+### 3. Install Optional Dependencies
+
+#### Wallet Dependencies
+
+It is not necessary to build wallet functionality to run bitcoind or the GUI. To enable legacy wallets, you must install `db4`. To enable [descriptor wallets](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md), `sqlite3` is required. Skip `db4` if you intend to *exclusively* use descriptor wallets
+
+###### Legacy Wallet Support
+
+`db4` is required to enable support for legacy wallets. Skip if you don't intend to use legacy wallets
 
-### Building BerkeleyDB
+```bash
+pkgin install db4
+```
 
-BerkeleyDB is only necessary for the wallet functionality. To skip this, pass
-`--disable-wallet` to `./configure` and skip to the next section.
+###### Descriptor Wallet Support
 
-It is recommended to use Berkeley DB 4.8. You cannot use the BerkeleyDB library
-from ports, for the same reason as boost above (g++/libstd++ incompatibility).
-If you have to build it yourself, you can use [the installation script included
-in contrib/](/contrib/install_db4.sh) like so:
+`sqlite3` is required to enable support for descriptor wallets. Skip if you don't intend to use descriptor wallets.
 
 ```bash
-./contrib/install_db4.sh `pwd`
+pkgin install sqlite3
 ```
+---
+
+#### GUI Dependencies
+###### Qt5
 
-from the root of the repository. Then set `BDB_PREFIX` for the next section:
+Bitcoin Core includes a GUI built with the cross-platform Qt Framework. To compile the GUI, we need to install `qt5`. Skip if you don't intend to use the GUI.
 
 ```bash
-export BDB_PREFIX="$PWD/db4"
+pkgin install qt5
 ```
+###### qrencode
+
+The GUI can encode addresses in a QR Code. To build in QR support for the GUI, install `qrencode`. Skip if not using the GUI or don't want QR code functionality.
+
+```bash
+pkgin install qrencode
+```
+---
+
+#### Test Suite Dependencies
+
+There is an included test suite that is useful for testing code changes when developing.
+To run the test suite (recommended), you will need to have Python 3 installed:
+
+```bash
+pkgin install python37
```

Comment:
See comment above about the Python version.

Reply:
see: https://github.com/bitcoin/bitcoin/pull/20610#discussion_r672795788